edfio/edf.py

```diff
@@ -1327,12 +1328,13 @@ def _calculate_num_data_records(
         raise ValueError(
             f"data_record_duration must be positive, got {data_record_duration}"
         )
-    required_num_data_records = signal_duration / data_record_duration
-    if required_num_data_records != int(required_num_data_records):
-        raise ValueError(
-            f"Signal duration of {signal_duration}s is not exactly divisible by data_record_duration of {data_record_duration}s"
-        )
-    return int(required_num_data_records)
+    for f in (lambda x: x, lambda x: Decimal(str(x))):
```

Comment:
Why try both approaches? Wouldn't it be sufficient (and more correct) to use the decimal arithmetic anyway?

Reply:
That would fail for fractions that have a periodic decimal representation, due to the finite length of the string representation. Say we have a `signal_duration` of 1s and wanted to set the `data_record_duration` to 1/3, `num_data_records` would be calculated as 
```python
>>> from decimal import Decimal
>>> Decimal(str(1)) / Decimal(str(1/3))
Decimal('3.000000000000000300000000000')
```

Reply:
True, for some reason this works though :thinking:
```
>>> 1 / (1/3)
3.0
```

Yeah using a tolerance should work! Is that your preferred solution?

Reply:
Let's leave it as it is. The choice for the tolerance is difficult to argue, while the decimal trick is easily explained by the decimal encoding in the Edf header.